lib/lookup.js

```diff
@@ -77,7 +83,8 @@ function resolve(context, next) {
           getRepo(rep.repo, meta),
           rep.master ? null : detail.spec,
           meta['dist-tags'],
-          rep.master ? null : rep.prefix);
+          rep.master ? null : rep.prefix,
+          context.options.sha || rep.sha);
```

Comment:
would you be able to include a test that has coverage of the `context.options.sha` code path